### DISCUSSION TITLE
Note when reset to other branch commit

### DISCUSSION THREAD

**Discussion by LinWanCen** [gitextensions/gitextensions#10185] [Q&A]
### Feature description

Pop-up window note when reset to non-current branch commit.

For example,
my branch is 20220710, I want to reset to HEAD for branch 20220701,
but wrong reset to a commit for branch 20220810,
the files for local branch 20220710 will same to branch 20220810 files,
will trigger a branch version disaster.

### Environment

- Git Extensions 3.5.4.12724
- Build 65f01f39982ccb121c4df7d4b00b506939553958
- Git 2.33.0.windows.2
- Microsoft Windows NT 10.0.19044.0
- .NET Framework 4.8.4510.0
- DPI 96dpi (no scaling)

**Comment by RussKie** [comment]
Unfortunately, I am having difficulties parsing your request. Do you think you could provide some sort of screenshots or an animation of what you're observing and what would you like to happen? Thanks

**Comment by LinWanCen** [comment]
![image](https://user-images.githubusercontent.com/14926671/188893459-90f23a11-0dd2-4910-91ee-76b5d6980d49.png)
![image](https://user-images.githubusercontent.com/14926671/188894719-794c668c-6160-486f-abf6-8e9d8e253d22.png)

**Comment by mstv** [comment]
Perhaps you do not really want to `reset current branch`. Rather `checkout branch`? Or do you want to reset uncommited changes (button in Commit dialog)?

You can get back the commit by showing the RefLog references.

![image](https://user-images.githubusercontent.com/36601201/188927718-85bfdbb6-9b92-48a6-9232-0e24ef1d8e22.png)

**Reply by LinWanCen** [reply]
thanks. it help author to do local local recovery.
![image](https://user-images.githubusercontent.com/14926671/189000651-76483542-5f16-4de3-868d-a2a23abe7dde.png)

But the problem is that colleagues will push such to the remote, causing problems with our remote branches.

**Comment by gerhardol** [comment]
Git is made to allow manipulation in many ways. If you add yet another popup for this, there are many other operations that can give similar effects. To manually implement such checks can be limiting for those that use it. (I do for my work branches).

The Git server should be setup to reject operations on protected branches.

**Reply by pmiossec** [reply]
If I understand well @LinWanCen request, he wants to have a specific warning message when the reset is done on a commit that doesn't belong to the current branch.
It could make sense from a user perspective but from a git perspective a lot less due to way Git is handling branches as pointer on commits.
So in practice it will be not so easy to develop.

**Comment by RussKie** [comment]
I think in this situation the best way to address the issue is via a user
training. Git hardly provides any guardrails, Git Extensions provides some
guardrails, but ultimately the user needs to understand the process and
bare responsibilities for the actions. We cannot know that instead of `git
reset` the user wanted to do `git checkout`.

**Comment by LinWanCen** [comment]
I found it have red text, Another pop-up window is better.
![image](https://user-images.githubusercontent.com/14926671/192072780-e547f1c6-b338-4851-ac20-3ea573c70347.png)

**Reply by vbjay** [reply]
https://github.com/gitextensions/gitextensions/discussions/10185#discussioncomment-3601346

Ruskie is right. This dialog is the chance for you to validate. Also using reflog you can reset back. So I don't think this is needed.

**Comment by vbjay** [comment]
https://hackernoon.com/reset-revert-and-checkout-in-git-5hdz3y6f

**Reply by vbjay** [reply]
https://learngitbranching.js.org/
